src/bp-blogs/bp-blogs-template.php

```diff
@@ -789,6 +790,45 @@ function bp_get_blog_latest_post( $args = array() ) {
 		return apply_filters( 'bp_get_blog_latest_post', $retval, $r );
 	}
 
+	/**
+	 * Output the ID of the latest post on the current blog in the loop.
+	 *
+	 * @since 14.3.4
+	 *
+	 * @see bp_get_blog_latest_post_id()
+	 */
+	function bp_blog_latest_post_id() {
+		echo esc_html( bp_get_blog_latest_post_id() );
+	}
+
+	/**
+	 * Return the ID of the latest post on the current blog in the loop.
+	 *
+	 * @since 14.3.4
+	 *
+	 * @global BP_Blogs_Template $blogs_template The main blog template loop class.
+	 *
+	 * @return string Post ID.
+	 */
+	function bp_get_blog_latest_post_id() {
+		global $blogs_template;
+
+		$retval = '';
+
+		if ( ! empty( $blogs_template->blog->latest_post ) && ! empty( $blogs_template->blog->latest_post->ID ) ) {
+			$retval = $blogs_template->blog->latest_post->ID;
+		}
+
+		/**
+		 * Filters the title text of the latest post on the current blog in the loop.
+		 *
+		 * @since 1.7.0
+		 *
+		 * @param string $retval Title text for the latest post.
+		 */
+		return apply_filters( 'bp_get_blog_latest_post_id', $retval );
```

Comment:
It shouldn't be needed, but maybe adding a cast to integer would make everyone feel better here. Like:
`return apply_filters( 'bp_get_blog_latest_post_id', (int) $retval );`